File: Java/Data Structures & Algorithms Module - Java/Revision/BinarySearchTree4/Node.java
/*
Node.java
Billy Cussen
01/02/2021
*/

public class Node<T>{

    T element;
    Node<T> left;
    Node<T> right;

    public Node(T element){
        this.element = element;
        this.left = null;
        this.right = null;
    }

    @Override
    public String toString(){
        String left = this.left==null ? "NULL" : this.left.element.toString();
        String right = this.right == null ? "NULL" : this.right.element.toString();
        return "Current: "+this.element+", Left: "+left+", Right: "+right;
    }

}

File: Python/Data Structures & Algorithms Module - Python/Revision/BST2/Node.py
"""
Node.py
Billy Cussen
02/02/2021
"""

class Node:

    def __init__(self, element):
        self.element = element
        self.left = None
        self.right = None

    def toString(self):
        leftNode = "NULL" if self.left == None else str(self.left.element)
        rightNode = "NULL" if self.right == None else str(self.right.element)
        return "Current: "+str(self.element)+", Left: "+leftNode+", Right: "+rightNode
